packages/flutter/lib/src/material/button_theme.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Used with [ButtonTheme] and [ButtonThemeData] to define a button's base
+/// colors, and the defaults for the button's minimum size, internal padding,
+/// and shape.
+///
+/// See also:
+///
+///  * [RaisedButton], which styles itself based on the ambient [ButtonTheme].
+///  * [FlatButton], which styles itself based on the ambient [ButtonTheme].
+enum ButtonTextTheme {
+  /// Button text is black or white depending on [ThemeData.brightness].
+  normal,
+
+  /// Button text is [ThemeData.accentColor].
+  accent,
+
+  /// Button text is based on [ThemeData.primaryColor].
+  primary,
+}
+
+/// Used with [ButtonThemeData] to configure the color and geometry of buttons.
+///
+/// A button theme can be specified as part of the overall Material theme
+/// using [ThemeData.buttomTheme]. The Material theme's button theme data
+/// can be overridden with [ButtonTheme].
+///
+/// The actual appearance of buttons depends on the button theme, the
+/// button's enabled state, its elevation (if any) and the overall Material
+/// theme.
+///
+/// See also:
+///
+///  * [FlatButton] and [RaisedButton], which are styled based on the
+///    ambient button theme.
+///  * [ThemeData.textTheme], `button` is the default text style for button labels.
+///  * [ThemeData.buttonColor], the fill color for [RaisedButton]s unless the
+///    button theme's text theme is [ButtonTextTheme.primary].
+///  * [ThemeData.primaryColor], the fill or text color if a button theme's text
+///    theme is [ButtonTextTheme.primary].
+///  * [ThemeData.accentColor], the text color for buttons when button theme's
+///    text theme is [ButtonTextTheme.accent].
+///  * [ThemeData.disabled], the default text color for disabled buttons.
+///  * [ThemeData.brightness], used to select contrasting text and fill colors.
+///  * [ThemeData.highlightColor], a button [InkWell]'s default highlight color.
+///  * [ThemeData.splashColor], a button [InkWell]'s default splash color.
+///  * [RawMaterialButton], which can be used to configure a button that doesn't
+///    depend on any inherited themes.
+class ButtonTheme extends InheritedWidget {
+  /// Creates a button theme.
+  ///
+  /// The [textTheme], [minWidth], and [height] arguments must not be null.
+  ButtonTheme({
```

Comment:
This used to be a const constructor.  Is it intentional that it's not anymore?  (the change broke some clients inside Google because they were calling `const ButtonTheme(...)`)

Reply:
Looks like it's the construction of `data` that necessitated this.  The same thing happened in `ButtonTheme.bar`.  @Hixie I'm thinking it's not worth sending out a breaking change notification since the fix is so obvious.  You agree?

Reply:
FYI, I updated the changelog to call out these changes.

https://github.com/flutter/flutter/wiki/Changelog#api-changes-to-buttontheme

Reply:
I've sent out a breaking change notification.